src/Chemistry.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef CHEMISTRY_HPP_ // NOLINT
+#define CHEMISTRY_HPP_
+//==============================================================================
+// TwoMomentRad - a radiation transport library for patch-based AMR codes
+// Copyright 2020 Benjamin Wibking.
+// Released under the MIT license. See LICENSE file included in the GitHub repo.
+//==============================================================================
+/// \file CloudyCooling.hpp
+/// \brief Defines methods for interpolating cooling rates from Cloudy tables.
+///
+
+#include <limits>
+
+#include "AMReX.H"
+#include "AMReX_BLassert.H"
+#include "AMReX_Extension.H"
+#include "AMReX_GpuQualifiers.H"
+
+#include "hydro_system.hpp"
+#include "radiation_system.hpp"
+
+#include "burn_type.H"
+#include "burner.H"
+#include "eos.H"
+#include "extern_parameters.H"
+
+namespace quokka::chemistry
+{
+template <typename problem_t> void computeChemistry(amrex::MultiFab &mf, const Real dt_in)
+{
+	const Real dt = dt_in;
+
+	const auto &ba = mf.boxArray();
+	const auto &dmap = mf.DistributionMap();
```

Comment:
Can you remove `ba` and `dmap`? They were only used to initialize `nsubstepsMF`, which is deleted now.